dotnet/src/Connectors/Connectors.Memory.PgVector/PostgresDbClient.cs

```diff
@@ -37,12 +37,12 @@ internal sealed class PostgresDbClient(NpgsqlDataSource dataSource, string? sche
 
     public void Dispose()
     {
-        // An instance of PostgresDbClient can be shared between a single store and multiple collections.
-        // The reference count is used to track how many collections are using this instance.
-        // When the number gets to zero, the DataSource is getting disposed.
-        if (Interlocked.Decrement(ref this._referenceCount) == 0)
+        if (this._ownsDataSource)
         {
-            if (this._ownsDataSource)
+            // An instance of PostgresDbClient can be shared between a single store and multiple collections.
+            // The reference count is used to track how many collections are using this instance.
+            // When the number gets to zero, the DataSource is getting disposed.
+            if (Interlocked.Decrement(ref this._referenceCount) == 0)
```

Comment:
This comment explains the approach I took for dealing with premature disposal.

From the user perspective, they can dispose the store/collection as soon as they are done using it without the need to keep certain order (like dispose the store only after the collections it created got disposed)

From our perspective, the implementation is thread safe and fast (the reference count is `Int32` rather than `Int64`, so on 32 bit architectures like x86 or arm32 there will be no need for more expensive operations required for atomic updates of 64 bit values)